align/compiler/write_verilog_lef.py

```diff
@@ -469,7 +469,13 @@ def WriteCap(graph,input_dir,name,unit_size_cap,all_array):
                 blocks_in_line = [blocks[blocks.find("{")+1:blocks.find("}")] for blocks in line.split(' , ') if ',' in blocks]
                 logger.info("place symmetrical cap as CC:%s",blocks_in_line)
                 for pair in blocks_in_line:
-                    p1,p2=pair.split(',')
+                    if ',' in pair:
+                        p1,p2=pair.split(',')
+                    elif '|' in pair:
+                        p1,p2=pair.split('|')
+                    else:
+                        logger.error(f"Neither ',' nor '|' found in {pair}.")
+                        assert False, pair
```

Comment:
Looks good overall but | is used as a scope operator for hierarchy as well. The vocabulary is not very clear.

@kkunal1408: Why do we have lines like this? (picked from linear_equalizer.const)
SymmBlock ( {xI0|MN0,xI1|MN0} , {R3,R4} , {xI4|MN0_xI2|MN0_xI3|MN0} , {MN6} , {MN9} )

Does the ',' in  {xI0|MN0,xI1|MN0} mean something different from the '|' in {xI4|MN0_xI2|MN0_xI3|MN0} ?

Line 469 looks suspect as well under this syntax.
                blocks_in_line = [blocks[blocks.find("{")+1:blocks.find("}")] for blocks in line.split(' , ') if ',' in blocks]
The ',' within {} would have caused a split already

Reply:
the '|' is added as a hierarchy operator when the designs are flattened. 
Underscore '_' I add while adding two or more blocks, e.g., m0 and m1 if make a differential pair I reduce the graph (remove m0 and m1) and add a node called m0_m1.
xI0|MN0 --> MN0 within xI0 hierarchy.
xI4|MN0_xI2|MN0_xI3|MN0 -->  It is a combination of three blocks xI4|MN0, xI2|MN0,
xI3|MN0 belonging to hierarchies of xI4, xI2 and xI3 respectively.

The split based on '|' shouldn't be done here. This part of code is to make any capacitor pair which is assigned Symmblock constraint as common centroid constraint.
I am seeing some different error. Let me debug and update. 

I ran pytest locally with a timeout. Where does it store the errors/log?

pytest --timeout=300 -vv --runnightly -k 'not SAR_ADC_6b_2GSPS and not vco and not sc_dc_dc_converter' -- tests/integration

Reply:
You can find the DRC errors in $ALIGN_WORK_DIR/<fubname>/3_pnr/*.errors.
If there is no .errors file, then that means there was a failure before a layout was generated. I would just rerun the test and look at the log generated when the test is being run.
```bash
pytest -vv --runnightly -k '<substring of the test name>' -- tests/integration
```

Reply:
Or you can scan through the junit.xml file.

Reply:
think this can be abandoned but I will let @kkunal1408 confirm.